src/runtime.rs

```diff
@@ -307,6 +328,20 @@ impl Class {
 
     }
 
+    /// Checks whether this class conforms to the specified protocol.
+    pub fn conforms_to_protocol(&self, proto: &Protocol) -> bool {
+        unsafe { class_conformsToProtocol(self, proto) == YES }
+    }
+
+    /// Get a list of the protocols to which this class conforms.
+    pub fn protocols(&self) -> MallocBuffer<&Protocol> {
```

Comment:
Hmm interesting conundrum, what should we call a method that wraps `protocol_copyProtocolList`?

This version for classes is `Class::protocols(&Self)`, but if we tried to add `Protocol::protocols(&Self)` it'd conflict with the `Protocol::protocols()` method. However, the symmetry between `Class::classes()` and `Protocol::protocols()` is nice...

What if we rename this to `Class::adopted_protocols(&Self)`? Then we're free to in the future add `Protocol::adopted_protocols(&Self)` which will parallel it nicely. And we still get to keep the nice symmetry between `Class::classes()` and `Protocol::protocols()`.